trunner/harness/psh.py

```diff
@@ -69,6 +77,11 @@ def __call__(self, result: TestResult) -> TestResult:
                 output=self.dut.before,
             ) from e
 
+        # Suppress klog output to console while test is running to avoid problems with parsing
+        if self.suppress_dmesg:
+            self.dut.pexpect_proc.sendline("dmesg -D")
+            self.dut.expect_exact(self.prompt)
```

Comment:
missing `prompt_timeout support` and we should translate `pexpect` exceptions to `ShellError`

Make waiting for prompt a separate function of this class and call it in all places needed